bot/cogs/random_cog/random_cog.py

```diff
@@ -133,5 +136,37 @@ async def slotsrolling(input, spinstatus,waittime):
         embed = await slotsrolling(f'{a} | {b} | {c}', f'**{message}**',1.75)
         await msg.edit(embed = embed)
 
+    @commands.command()
+    async def raffle(self, ctx, time: typing.Optional[Duration] = 5, *, reason):
+        """
+        Raffle command, picks a random winner from users who reacted with :tickets:
+        :param time: optional, timer for the raffle, default is 5 seconds
+        :param reason: required, reason/purpose for the raffle
+        """
+        if isinstance(time, datetime):
+            delay_time = (time - datetime.utcnow()).total_seconds()
+        else:
+            delay_time = time
+
+        description = f'Raffle for {reason}\nReact with :tickets: to enter the raffle'
+        embed = discord.Embed(title = 'RAFFLE', color=Colors.ClemsonOrange, description = description)
+        msg = await ctx.send(embed = embed)
+        await msg.add_reaction('🎟️')
+        await asyncio.sleep(delay_time)
+
+        cache_msg = await ctx.fetch_message(msg.id)
+        for reaction in cache_msg.reactions:
+            if reaction.emoji == '🎟️':
+                if reaction.count == 1:
+                    await ctx.send('No one entered the raffle :(')
```

Comment:
we should probably make this one an embed edit as well, in my testing it was kinda odd if there were messages between the raffle embed and the announcement no one entered